src/pages/index.js

```diff
@@ -6,46 +6,83 @@ import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import Section from "../components/Section.js";
 import UserInfo from "../components/UserInfo.js";
+import ProfilePicEvt from "../components/ProfilePicEvt.js";
 import {
-  initialCards,
   config,
   editProfileButton,
+  editProfilePictureButton,
   addCardButton,
+  profilePicture,
 } from "../utils/constants.js";
+import { api } from "../components/Api.js";
+let gallery;
 //Add Card Popup Form
 const addCardPopupForm = new PopupWithForm("#add-card-modal", {
   handleSubmitEvent: (inputValues) => {
-    const newCard = {
-      name: inputValues.name,
-      link: inputValues.bio,
-    };
-    const cardElement = createCard(newCard);
-    gallery.addItem(`prepend`, cardElement);
+    addCardPopupForm.renderLoading(true);
+    api
+      .addCard(inputValues)
+      .then((inputValues) => {
+        const cardElement = createCard(inputValues);
+        gallery.addItem(`prepend`, cardElement);
+        addCardPopupForm.renderLoading(false);
+      })
+      .catch((err) => {
+        console.log(err);
+        addCardPopupForm.renderLoading(false);
+      });
     addCardPopupForm.form.reset();
   },
 });
 // Edit Popup Form
 const editPopupForm = new PopupWithForm("#edit-profile-modal", {
   handleSubmitEvent: (inputValues) => {
-    userInfo.setUserInfo({ name: inputValues.name, job: inputValues.bio });
+    editPopupForm.renderLoading(true);
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```